### PR TITLE
Fix bug in memory usage calculation (unintended integer division)

### PR DESCRIPTION
**Pull request by practicalswift**
Fix bug in memory usage calculation (unintended integer division).

### CODE REVIEW

**Comment by promag** [comment]
ACK 490e8be.

File: src/dbwrapper.cpp [hunk 1]
```diff
@@ -159,12 +159,12 @@ bool CDBWrapper::WriteBatch(CDBBatch& batch, bool fSync)
     const bool log_memory = LogAcceptCategory(BCLog::LEVELDB);
     double mem_before = 0;
     if (log_memory) {
-        mem_before = DynamicMemoryUsage() / 1024 / 1024;
+        mem_before = (double)DynamicMemoryUsage() / 1024 / 1024;
```
Comment (laanwj) [inline src/dbwrapper.cpp]: Wouldn't replacing 1024 with 1024.0 here (and in the other case) have the same effect, but avoid the cast?

Comment (practicalswift) [reply src/dbwrapper.cpp]: @laanwj Absolutely! Now updated. Please re-review :-)

**Comment by eklitzke** [comment]
ACK a16c6d23cadb517f1e728c719a316768ae7b7fe8

I'm really tempted to add formatters for MiB and whatnot as a new format specifier in `tinyformat.h`, as we have this logic all over the place.

**Comment by MarcoFalke** [comment]
utACK a16c6d23cadb517f1e728c719a316768ae7b7fe8

**Review by eklitzke — APPROVED** [review]

**Comment by practicalswift** [comment]
@eklitzke Formatters for MiB, etc sounds nice!

**Comment by fanquake** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/12784/commits/a16c6d23cadb517f1e728c719a316768ae7b7fe8
I've kickstarted Travis.

**Comment by laanwj** [comment]
> I'm really tempted to add formatters for MiB and whatnot as a new format specifier in tinyformat.h, as we have this logic all over the place.

I'd prefer not to make such changes to tinyformat.h locally. Unless you manage to upstream it (https://github.com/c42f/tinyformat), an utility function would be preferable. Although this is arguably so simple I'm not sure how much is gained by that.

**Pull request merged**